src/submission/models.py

```diff
@@ -949,6 +952,8 @@ def current_stage_url(self):
             return reverse('proofing_article', kwargs=kwargs)
         elif self.stage == STAGE_READY_FOR_PUBLICATION:
             return reverse('publish_article', kwargs=kwargs)
+        else:
+            return reverse(self.current_stage.jump_url, kwargs=kwargs)
```

Comment:
all of the above branches can go now, except for `STAGE_UNASSIGNED` ( we could use a comment explaining that there is no `workflow_element` for the unassigned stage.